### ISSUE TITLE
Spring Boot fails to configure Liquibase with Honeycomb for jdk17 

### ISSUE THREAD

**Issue by giorgimode** [honeycombio/beeline-java#201]
**Versions**

- Java: 17.0.2
- beeline-spring-boot-starter: 2.0.0
- liquibase: 4.9.1

**Steps to reproduce**

1. Start Spring Boot with liquibase and honeycomb enabled with java 17.  Following error is thrown

> exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [liquibase.integration.spring.SpringLiquibase]: Factory method 'liquibase' threw exception; nested exception is org.springframework.boot.jdbc.UnsupportedDataSourcePropertyException: Unable to find suitable method for url

**Additional context**
After trying to trace the issue, I noticed when Honeycomb is enabled, LiquibaseAutoConfiguration autowires ProxyDataSource. When it's disabled, HikariDataSource is autowired correctly. I don't know how exactly honeycomb affects the application context.

Additionally, I was forced to specify hibernate-validator version. It's not necessary without honeycomb.

**Comment by giorgimode** [comment]
@robbkidd it's been working alright. I just upgraded Java from 11 to 17 and beeline-spring-boot-starter to the latest 2.0.0

**Comment by giorgimode** [comment]
Hi @MikeGoldsmith, thanks for looking into it since this bug forces us to disable Honeycomb in our system for now.

**Comment by MikeGoldsmith** [comment]
@giorgimode btw - you could disable JDBC instrumentation using the `honeycomb.beeline.jdbc.enabled=false` system property. It may help unblock you while we work on a permanent fix.

**Comment by giorgimode** [comment]
@MikeGoldsmith that is helpful. Application starts up normally with that flag. Thanks.

**Comment by JamieDanielson** [comment]
Hi @giorgimode, we've done some research but have not found a great solution for this yet. This will remain on our backlog for review but it may be a while before we can get to it, as we are currently focusing our efforts on our [OpenTelemetry Distro for Java](https://github.com/honeycombio/honeycomb-opentelemetry-java). I just want to ensure we set appropriate expectations. We are also open to PRs if this is something you are interested in taking on.

**Comment by giorgimode** [comment]
Thanks for the update, @JamieDanielson. For now I'll keep jdbc disabled

**Comment by JamieDanielson** [comment]
Hello,

We are continuing to trim our OSS backlog. Upon further review of this issue, we are seeing that there is still that viable workaround of disabling the instrumentation. We are closing this issue as it is a low priority for us to fix. It is unlikely that we'll ever get to it, and so we'd like to set expectations accordingly.

**Comment by giorgimode** [comment]
@JamieDanielson it's a shame that we're forced to disable jdbc instrumentation permanently. That's a big loss as observability goes. At least now I know. We'll look for alternatives

**Comment by JamieDanielson** [comment]
Understood, disabling instrumentation is not an ideal long-term scenario. We do recommend switching from Beeline to OpenTelemetry, which is where our efforts are focused and where more community members are contributing. Our [Java Distro](https://github.com/honeycombio/honeycomb-opentelemetry-java) is intended to help streamline configuration, and uses the upstream agent and SDK. The OTel Java Agent includes jdbc instrumentation [along with many other libraries and frameworks](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks) out of the box.

**Comment by giorgimode** [comment]
Thanks for the update. Are there any plans to provide spring honeycomb integration with opentelemetry? I cannot find any documentation so far.

**Comment by JamieDanielson** [comment]
Sorry I'm not sure I understand what you mean by "spring honeycomb integration"? OpenTelemetry has instrumentation for Spring, and that telemetry can be [sent to Honeycomb](https://docs.honeycomb.io/getting-data-in/opentelemetry/java-distro/#automatic-instrumentation-with-the-opentelemetry-java-agent). Does it help to review the [spring examples](https://github.com/honeycombio/honeycomb-opentelemetry-java/tree/main/examples) we have in our distro?

**Comment by giorgimode** [comment]
With `beeline-spring-boot-starter` dependency and adding few properties in Spring, you get full power of honeycomb. Would that kind of integration be added anytime soon? 
But [this Java example ](https://github.com/honeycombio/honeycomb-opentelemetry-java/blob/main/examples/spring-sdk/src/main/java/io/honeycomb/examples/springbootsdk/Application.java)should suffice

**Comment by robbkidd** [comment]
I love the phrase "full power of honeycomb" but I'm not certain what specific features or behavior you mean.

Honeycomb has a distribution of OpenTelemetry for Java that comes in two flavors: an **SDK** for manual instrumentation or a **Java Agent** (that includes the SDK) for auto- and manual instrumentation.

The Honeycomb distribution adds a few things to the upstream OpenTelemetry for Java:

* a streamlined configuration for sending data to Honeycomb
* a deterministic sampler similar to the one in the Beelines
* "[multi-span attributes](https://docs.honeycomb.io/getting-data-in/opentelemetry/java-distro/#multi-span-attributes)" which are the closest OpenTelemetry can currently get to Beeline "trace-level attributes"

You linked to the "spring-sdk" example which is an example of _only_ manual/custom instrumentation. This GitHub issue was started around auto-instrumentation, so my sense is that you will be more interested in the OpenTelemetry Java Agent which is where auto-instrumentation has been implemented. If that is true, I recommend following the [Quick Start](https://docs.honeycomb.io/getting-started/quickstart/#step-3-instrument-your-application-to-send-telemetry-data-to-honeycomb) guide for Java and referring to the [spring-agent-manual](https://github.com/honeycombio/honeycomb-opentelemetry-java/tree/main/examples/spring-agent-manual) example application instead for a mix of auto-instrumentation and custom manual telemetry.

**Comment by giorgimode** [comment]
Auto-instrumentation is what I was looking for. My goal is to avoid polluting my source with tracing code as much as possible. Using an agent and environment variables accomplishes that. I can add aop for special cases but by default I'd rather rely on auto-instrumentation. It's much clearer now, thanks.
